kategory-docs/docs/docs/patterns/glossary/README.md

```diff
@@ -0,0 +1,145 @@
+---
+layout: docs
+title: Functional Programming Glossary
+permalink: /docs/patterns/glossary/
+---
+
+## Functional Programming Glossary
+
+TODO: expand terms and usage
+
+### Datatypes
+
+A datatype is a class that encapsulates one reusable coding pattern.
+These solutions have a canonical implementation that is generalised for all possible uses.
+
+Some common patterns expressed as datatypes are absence handling with [`Option`]({{ '/docs/datatypes/option' | relative_url }}),
+branching in code with [`Either`]({{ '/docs/datatypes/either' | relative_url }}),
+catching exceptions with [`Try`]({{ '/docs/datatypes/try' | relative_url }}),
+or interacting with the platform the program runs in using [`IO`]({{ '/docs/effects/io' | relative_url }}).
+
+### Typeclasses
+
+A typeclass is an interface representing one behavior associated with a type.
+Examples of this behavior are comparison ([`Eq`]({{ '/docs/typeclasses/eq' | relative_url }})),
+composability ([`Monoid`]({{ '/docs/typeclasses/monoid' | relative_url }})),
+its contents are mappable ([`Functor`]({{ '/docs/typeclasses/functor' | relative_url }})),
+or error recovery ([`MonadError`]({{ '/docs/typeclasses/monaderror' | relative_url }})).
+
+```kotlin
+interface Eq<F>: Typeclass {
+  fun eqv(a: F, b: F): Boolean
+}
+```
+
+What differentiates typeclasses from regular interfaces is that they are meant to be created at a global scope for a single type.
```

Comment:
> that they are meant to be created at a global scope for a single type.

Aren't they meant for defining behavior outside of a class definition? Global scope typeclasses is a characteristic for Kategory.

Reply:
Typeclasses are [officially recognized](https://wiki.haskell.org/OOP_vs_type_classes#Type_classes_are_like_interfaces.2Fabstract_classes.2C_not_classes_itself) to be very similar to interfaces by Haskell.org. 

So I am not sure about this differentiation either. I mean, interfaces with generic parameters just the most accurate way to represent type classes that support ad-hoc polymorphism on languages like Java, Kotlin or C#. That means Typeclass and Interface are two ortogonal concepts that should probably not be compared.